diff --git a/metal/v1/model_auth_token_project.go b/metal/v1/model_auth_token_project.go
index 678b073..e813e4a 100644
--- a/metal/v1/model_auth_token_project.go
+++ b/metal/v1/model_auth_token_project.go
@@ -13,6 +13,7 @@ package v1
 
 import (
 	"encoding/json"
+	"time"
 )
 
 // AuthTokenProject struct for AuthTokenProject
diff --git a/metal/v1/model_auth_token_user.go b/metal/v1/model_auth_token_user.go
index 5328a37..5b2ae4f 100644
--- a/metal/v1/model_auth_token_user.go
+++ b/metal/v1/model_auth_token_user.go
@@ -13,6 +13,7 @@ package v1
 
 import (
 	"encoding/json"
+	"time"
 )
 
 // AuthTokenUser struct for AuthTokenUser
diff --git a/metal/v1/model_device_create_in_facility_input.go b/metal/v1/model_device_create_in_facility_input.go
index 5101ee5..d3f56fe 100644
--- a/metal/v1/model_device_create_in_facility_input.go
+++ b/metal/v1/model_device_create_in_facility_input.go
@@ -13,6 +13,7 @@ package v1
 
 import (
 	"encoding/json"
+	"time"
 )
 
 // DeviceCreateInFacilityInput struct for DeviceCreateInFacilityInput
diff --git a/metal/v1/model_device_create_in_metro_input.go b/metal/v1/model_device_create_in_metro_input.go
index 4027d6b..d7c2b4a 100644
--- a/metal/v1/model_device_create_in_metro_input.go
+++ b/metal/v1/model_device_create_in_metro_input.go
@@ -13,6 +13,7 @@ package v1
 
 import (
 	"encoding/json"
+	"time"
 )
 
 // DeviceCreateInMetroInput struct for DeviceCreateInMetroInput
diff --git a/metal/v1/model_device_created_by.go b/metal/v1/model_device_created_by.go
index 8ad0bc0..26318ff 100644
--- a/metal/v1/model_device_created_by.go
+++ b/metal/v1/model_device_created_by.go
@@ -13,6 +13,7 @@ package v1
 
 import (
 	"encoding/json"
+	"time"
 )
 
 // DeviceCreatedBy struct for DeviceCreatedBy
diff --git a/metal/v1/model_instances_batch_create_input_batches_inner.go b/metal/v1/model_instances_batch_create_input_batches_inner.go
index f49cd81..4e31e33 100644
--- a/metal/v1/model_instances_batch_create_input_batches_inner.go
+++ b/metal/v1/model_instances_batch_create_input_batches_inner.go
@@ -13,6 +13,7 @@ package v1
 
 import (
 	"encoding/json"
+	"time"
 )
 
 // InstancesBatchCreateInputBatchesInner struct for InstancesBatchCreateInputBatchesInner
